pkg/provisioner/ironic/factory.go

```diff
@@ -157,6 +157,15 @@ func loadConfigFromEnv(havePreprovImgBuilder bool) (ironicConfig, error) {
 
 	// Let's see if externalURL looks like a URL
 	if c.externalURL != "" {
+		podIPs := strings.Split(os.Getenv("POD_IPS"), ",")
+		for _, podIP := range podIPs {
+			if strings.Contains(podIP, ":") {
+				// IPv6 found, try to substitute the external URL
+				c.externalURL = strings.Replace(c.externalURL, "(POD_IP)", fmt.Sprintf("[%s]", podIP), 1)
```

Comment:
Can we make _this_ bit a little less hacky?

```
v6URL, externalURLParseErr := url.Parse(c.externalURL)
...
v6URL.Host = net.JoinHostPort(podIP, v6URL.Port())
c.externalURL = v6URL.String()
```

Bonus: no dependency on openshift/cluster-baremetal-operator#381

Reply:
We will need a CBO change to supply something, although that can be localhost, of course.